### PR TITLE
Use futures 0.3 API

### PR DESCRIPTION
**Pull request by bartlomieju**
This is #3299 by @afinch7 - I'm just trying to get it to green

### CODE REVIEW

**Comment by bartlomieju** [comment]
I had to disable `[http] serveTLS` possibly related to #3160

**Comment by bartlomieju** [comment]
@ry @afinch7 this should be now landable - I had to disable only 1 test (standard lib, `[http] serveTLS`). I think we should land this PR and start upgrading dependencies as they switch to new futures API.

File: cli/file_fetcher.rs [hunk 1]
```diff
@@ -119,14 +122,14 @@ impl SourceFileFetcher {
   pub fn fetch_source_file_async(
     self: &Self,
     specifier: &ModuleSpecifier,
-  ) -> Box<SourceFileFuture> {
+  ) -> Pin<Box<SourceFileFuture>> {
```
Comment (ry) [inline cli/file_fetcher.rs]: What is the Pin for?

File: cli/ops/io.rs [hunk 2]
```diff
@@ -8,11 +8,16 @@ use deno::ErrBox;
 use deno::Resource;
 use deno::*;
 use futures;
-use futures::Future;
-use futures::Poll;
+use futures::compat::AsyncRead01CompatExt;
+use futures::compat::AsyncWrite01CompatExt;
```
Comment (ry) [inline cli/ops/io.rs]: It's very helpful that they've included this.

File: cli/compilers/ts.rs [hunk 3]
```diff
@@ -270,19 +272,17 @@ impl TsCompiler {
...
-          eprintln!("{}", err.to_string());
-          std::process::exit(1);
-        }
-        debug!("Sent message to worker");
-        worker_.get_message()
-      });
+    let first_msg_fut = async move {
+      worker.post_message(req_msg).await.unwrap();
+      let result = worker.await;
+      if let Err(err) = result {
```
Comment (ry) [inline cli/compilers/ts.rs]: Nice!

**Pull request merged**